### PR TITLE
[feat]:Auth

### PR DESCRIPTION
**Pull request by ultronjs**
Feature added in this PR
1.)Auth Logic and Context
2.)SignIn Page UI and Logic
3.)SignUp Page UI and Logic
4.)Added Toast for error and success

### CODE REVIEW

File: src/index.css [hunk 1]
```diff
@@ -11,3 +13,97 @@ code {
...
+.form_container_img{
+    position: absolute;
+    z-index: -1;
+    top: 0;
+    bottom: 0;
+    left: 0;
+    right: 0;
+    width: 100%;
+    height: 100%;
+  /* height: 100%;
```
Comment (sameep-s) [inline src/index.css:40]: you can remove these comments.

File: src/context/authContext.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    const initialSignInStatus = {
+      status: localStorage.getItem("token")===null?false:true,
+      encodedToken: localStorage.getItem("token"),
+    };
+
+
+    const signInReducer = (signInStatus,action) => {
+
+        switch(action.type){
+            case "SET_USER":
```
Comment (sameep-s) [inline src/context/authContext.js:15]: you can create constants for these actions and import and use them everywhere this is a good practice and will remove alot of typos.

**Review by sameep-s — APPROVED** [review]
You can use Constants for reducer actions to minimize typos. Otherwise code looks good you can merge.

File: src/index.css (hunk 1 above)
Comment (ultronjs) [reply src/index.css:40]: Removed

File: src/context/authContext.js (hunk 2 above)
Comment (ultronjs) [reply src/context/authContext.js:15]: Will do this in refactoring

**Review by JantuDeb — APPROVED** [review]
Looks good to me

**Comment by ultronjs** [comment]
Merging after 2 approvals

**Pull request merged**